src/NHibernate/Transform/AliasToBeanResultTransformer.cs

```diff
@@ -269,9 +276,19 @@ public NamedMember(string name, T[] members)
 				}
 			}
 
-			public string Name;
-			public T Member;
-			public T[] AmbiguousMembers;
+			private NamedMember(SerializationInfo info, StreamingContext context)
+			{
+				Name = info.GetString("Name");
+				Member = info.GetValue<T>("Member");
+				AmbiguousMembers = info.GetValueArray<T>("AmbiguousMembers");
```

Comment:
Is that custom helper really needed? What is the issue in serializing directly an array? (It seems this helper was no more having any usages and could have been obsoleted, along with its add counterpart.)

Reply:
Do you mean `GetValueArray`?

Reply:
Yes.